extension/src/extension.ts

```diff
@@ -59,54 +61,42 @@ export function activate(context: vscode.ExtensionContext) {
         });
     });
 
-    const getCommandToRunSpy = (serverPath) => {
-        const os = execSync("uname -s").toString().toLowerCase();
-        if (os.indexOf("darwin") !== -1) {
-            return `osascript -e 'tell application "Terminal" to do script "node ${serverPath}"'`;
-        } else if (os.indexOf("linux") !== -1) {
-            return `xterm -e node ${serverPath}`;
+    const getServerUrl = () => {
+        const baseName = path.basename(vscode.workspace.rootPath);
+        const configurationFile = `${vscode.workspace.rootPath}${path.sep}Configuration.${baseName}${path.sep}Dev${path.sep}${baseName}_Deployment_Configuration.xml`;
+        let serverUrl = undefined;
+        if (fs.existsSync(configurationFile)) {
+            const json = parseStringSync(fs.readFileSync(configurationFile).toString());
+            const websocketConfig = json.deployment.configuration[0].gateways[0].websocket[0];
+            const s = (websocketConfig.$.type === "Secure") ? "s" : "";
+            serverUrl = `ws${s}://${websocketConfig.$.host}:${websocketConfig.$.bridgeport}`;
         } else {
-            return `START cmd.exe /K node ${serverPath}`;
+            vscode.window.showWarningMessage(`File ${configurationFile} not found`);
         }
+        return serverUrl;
     };
 
     promisify(freeport)()
         .then((port: number) => {
             const disposableSpy = vscode.commands.registerCommand("xcomponent.launch.spy", () => {
-                const dirPath = path.parse(context.extensionPath);
-                const serverPath = `${dirPath.dir}${path.sep}spy${path.sep}server.js`;
-                const runSpyServercommand = getCommandToRunSpy(serverPath);
-                (<any>process.env.port) = port;
+                const serverUrl = getServerUrl();
+                const binPath = path.join(context.extensionPath, "out/spy/bin");
+                const runSpyServercommand = `webpack-dev-server --content-base ${binPath} --port ${port}`;
```

Comment:
Now we have a dependency to webpack-dev-server. why ?

Reply:
It is really a strong dependency. Nobody wants to install webpack-dev-server as global.
On the contrary, node or serve can be install as a global dependency